include/af/memory.h

```diff
@@ -0,0 +1,65 @@
+/*******************************************************
+ * Copyright (c) 2019, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <af/defines.h>
+#include <af/event.h>
+
+#if AF_API_VERSION >= 37
+
+typedef void* af_buffer_info;
+
+#ifdef __cplusplus
+namespace af {
+
+/// A simple RAII wrapper for af_buffer_info
+class buffer_info {
```

Comment:
This name kind of conflicts with the bufferInfo function in the internal memory manager. Could you rename that function to avoid confusion?